include/dmtr/fail.h

```diff
@@ -10,43 +10,44 @@
 #include <stdlib.h>
 
 #ifdef __cplusplus
-extern "C" {
+extern "C"
+{
 #endif
 
-typedef void (*dmtr_onfail_t)(int error_arg,
-      const char *expr_arg, const char *funcn_arg, const char *filen_arg,
-      int lineno_arg);
-
-#define DMTR_TRUE2(Error, Condition, ErrorCache) \
-    do { \
-        const int ErrorCache = (Error); \
-        if (DMTR_UNLIKELY(!(Condition))) { \
-            dmtr_fail(ErrorCache, #Condition, NULL, __FILE__, __LINE__);  \
-            return ErrorCache; \
-        } \
-   } while (0)
-
-#define DMTR_TRUE(Error, Condition) \
-    DMTR_TRUE2(Error, Condition, DMTR_TRUE_errorCache)
-
-#define DMTR_OK2(Error, ErrorCache) \
-    do { \
-        const int ErrorCache = (Error); \
-        if (DMTR_UNLIKELY(0 != ErrorCache)) { \
-            dmtr_fail(ErrorCache, #Error, NULL, __FILE__, __LINE__); \
-            return ErrorCache; \
-        } \
+    typedef void (*dmtr_onfail_t)(int error_arg, const char *expr_arg, const char *funcn_arg, const char *filen_arg,
```

Comment:
What style is this?  The only one I've consistently seen used inside Microsoft is: either all arguments must fit on the same line, or each argument must be on its own line.  At least for actual function calls, I don't know if this is normally enforced for typedefs of function calls.

Reply:
This is Microsoft's style, applied by clang formatting tool:
- https://clang.llvm.org/docs/ClangFormatStyleOptions.html
- https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-code-style-settings-reference

Let me know if you prefer any of the other styles.

Reply:
Hmm, not clear where that "Microsoft" style is actually coming from.  That page seems to only cover DotNet languages.  At least for this specific issue, it doesn't conform to what I'm used to seeing used internally.  I'll have to look over our options here.

The specific "Configurable Format Style Options" that appear to pertain to this issue are BinPackArguments and BinPackParameters, both of which should be false.  Although that doesn't completely cover it, as it still allows the first argument or parameter to appear on the same line as the function name.  I think AlignAfterOpenBracket would also need to be set to BAS_AlwaysBreak.

Reply:
I understand your concern. It should be awkward to have something named wrong this way.

Now, I would suggest us to stick to some built-in formatting style of clang, because it is easier for external contributors to match the standard that we choose to embrace. VIM, EMACs and VSCode have facilities on this direction.

For me, any of these built-in would be fine. Which one do you chose?

- LLVM A style complying with the [LLVM coding standards](https://llvm.org/docs/CodingStandards.html)
- Google A style complying with [Google’s C++ style guide](https://google.github.io/styleguide/cppguide.html)

- Chromium A style complying with [Chromium’s style guide](https://chromium.googlesource.com/chromium/src/+/refs/heads/main/styleguide/styleguide.md)
- Mozilla A style complying with [Mozilla’s style guide](https://firefox-source-docs.mozilla.org/code-quality/coding-style/index.html)

- WebKit A style complying with [WebKit’s style guide](https://www.webkit.org/coding/coding-style.html)
- Microsoft A style complying with [Microsoft’s style guide](https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-code-style-settings-reference)
- GNU A style complying with the [GNU coding standards](https://www.gnu.org/prep/standards/standards.html)

Reply:
@BrianZill any thoughts on this?